### PR TITLE
migrate check_rhel_version

### PR DESCRIPTION
**Pull request by jabia99**

### CODE REVIEW

File: test_suite/generic/test_generic.py [hunk 1]
```diff
@@ -151,6 +152,27 @@ def test_inittab_and_systemd(self, host):
...
+        release_file = 'redhat-release'
+        if host.system_info.distribution == 'fedora':
+            release_file = 'fedora-release'
+
+        with host.sudo():
+            package_release_version = host.check_output("sudo rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
+
+        assert product_version == package_release_version, \
+            f'product version ({product_version}) does not match package release version'
+        assert product_version.replace('.', '-') in cloud_image_name, 'product version in not in image name'
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:174]:
```suggestion
        assert str(product_version).replace('.', '-') in cloud_image_name, 'product version is not in image name'
```

File: test_suite/generic/test_generic.py [hunk 2]
```diff
@@ -151,6 +152,27 @@ def test_inittab_and_systemd(self, host):
...
+        product_version = host.system_info.release
+
+        release_file = 'redhat-release'
+        if host.system_info.distribution == 'fedora':
+            release_file = 'fedora-release'
+
+        with host.sudo():
+            package_release_version = host.check_output("sudo rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
+
+        assert product_version == package_release_version, \
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:172]:
```suggestion
        assert product_version == float(package_release_version), \
```

File: test_suite/generic/test_generic.py [hunk 3]
```diff
@@ -151,6 +152,27 @@ def test_inittab_and_systemd(self, host):
...
+
+        cloud_image_name = instance_data['name']
+        product_version = host.system_info.release
+
+        release_file = 'redhat-release'
+        if host.system_info.distribution == 'fedora':
+            release_file = 'fedora-release'
+
+        with host.sudo():
+            package_release_version = host.check_output("sudo rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:170]: No need to call sudo with sudo.

```suggestion
            package_release_version = host.check_output("rpm -q --qf '%{VERSION}' --whatprovides " + release_file)
```

Comment (narmaku) [inline test_suite/generic/test_generic.py:170]: I think there's a way to check package version with a testinfra module.

See: https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.package.Package.version

File: test_suite/generic/test_generic.py [hunk 4]
```diff
@@ -151,6 +152,27 @@ def test_inittab_and_systemd(self, host):
                     assert 'si::sysinit:/etc/rc.d/rc.sysinit' in host.check_output("grep '^si:' /etc/inittab"), \
                         'Unexpected default inittab "id"'
 
+    def test_release_version(self, host, instance_data):
+        '''
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:156]: Replace single quotes by double quotes for multi-line strings and docstrings. I think flake8 also highlights this.

File: test_suite/generic/test_generic.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 import pytest
-
+from pytest_mock import package_mocker
```
Comment (narmaku) [inline test_suite/generic/test_generic.py:2]: What is this ninja artifact?

**Review by narmaku — CHANGES_REQUESTED** [review]

File: test_suite/generic/test_generic.py (hunk 5 above)
Comment (jabia99) [reply test_suite/generic/test_generic.py:2]: upsss

File: test_suite/generic/test_generic.py (hunk 3 above)
Comment (jabia99) [reply test_suite/generic/test_generic.py:170]: Can't be used here as we are using --whatprovides option it's not available on test_infra

File: test_suite/generic/test_generic.py (hunk 1 above)
Comment (jabia99) [reply test_suite/generic/test_generic.py:174]: product version is already a string, I tested it and passed on all images

File: test_suite/generic/test_generic.py (hunk 2 above)
Comment (jabia99) [reply test_suite/generic/test_generic.py:172]: we are comparing string to string already. Would it be better comparing floats?

Comment (narmaku) [reply test_suite/generic/test_generic.py:172]: I see, and it's an "equals" comparison so it doesn't matter. Please dismiss this comment!

File: test_suite/generic/test_generic.py (hunk 3 above)
Comment (narmaku) [reply test_suite/generic/test_generic.py]: I see!

File: test_suite/generic/test_generic.py (hunk 1 above)
Comment (narmaku) [reply test_suite/generic/test_generic.py]: I see, it makes sense then to keep it like it was! Sorry about it.

**Review by narmaku — APPROVED** [review]
LGTM!

**Pull request merged**